syntax = "proto3";

// ===========================> USER PROTO FILE <===========================
// This file contains the proto file for the User Related Objects.
// ==========================================================================

enum Rarity { // --------------------------> ENUM FOR THE RARITY OF THE ACHIEVEMENT
    COMMON = 0;
    RARE = 1;
    EPIC = 2;
    LEGENDARY = 3;
    MYTHIC = 4;
}

message Achievement { // --------------------------> MESSAGE FOR ACHIEVEMENT
    string title = 1;
    string description = 2;
    string date = 3;
    Rarity rarity = 4;
}

message User { // --------------------------> MESSAGE FOR USER
    string user_name = 1;
    string password = 2;
    optional string location = 3;
    repeated string animes_watched = 4;
    repeated int32 anime_watched_score = 5;
    repeated string topics_subscribed = 6;
    int32 karma = 8;
    repeated Achievement achievements = 9;
}